src/amber/cli/recipes/recipe_fetcher.cr

```diff
@@ -0,0 +1,72 @@
+require "http/client"
+require "zip"
+
+module Amber::Recipes
+  class RecipeFetcher
+    getter kind : String
+    getter name : String
+    getter directory : String
+
+    def initialize(@kind : String, @name : String)
+      @directory = "#{Dir.current}/#{@name}/#{@kind}"
+    end
+
+    def fetch
+      # if the recipe exists in the local recipes folder then use that
+      if Dir.exists?(@directory)
+        return @directory
+      end
+
+      # if the given recipe name is a directory and has a sub directory
+      # for the kind of component being generated then use it
+      if Dir.exists?("#{@name}/#{@kind}")
+        return "#{@name}/#{@kind}"
+      end
+
+      # check the cached templates folder
+      template_path = "#{AMBER_RECIPE_FOLDER}/#{@name}"
+
+      if Dir.exists?("#{template_path}/#{@kind}")
+        return "#{template_path}/#{@kind}"
+      end
+
+      # otherwise fetch from the github repository
+      return fetch_url template_path
+    end
+
+    def fetch_url(template_path)
+      # download the recipe zip file from the github repository
+      HTTP::Client.get("https://raw.githubusercontent.com/AmberRecipes/recipes/master/#{@name}.zip") do |response|
```

Comment:
This base url should be a constant somewhere

Reply:
(Non Blocking) I will opt to make this URL configurable via the `.amber.yml` something like recipe sources. In case people would like to have their own private repo of recipes.